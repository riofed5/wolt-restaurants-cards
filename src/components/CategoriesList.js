import React from "react";
import chinese from "../images/thumbnails/cat_chinese.jpg";
import kebab from "../images/thumbnails/cat_kebab.jpg";
import ramen from "../images/thumbnails/cat_ramen.jpg";
import steak from "../images/thumbnails/cat_steak.jpg";
import streetfood from "../images/thumbnails/cat_street_food.jpg";
import sushi from "../images/thumbnails/cat_sushi.jpg";

const CategoriesList = () => {
  return (
    <div className="categories">
      <div className="categories-title">
        <span>Categories</span>
      </div>
      <div className="categories-center">
        <div className="thumbnails-container">
          <img src={kebab} alt="single tag" />
          <p>kebab</p>
        </div>
        <div className="thumbnails-container">
          <img src={ramen} alt="single tag" />
          <p>ramen</p>
        </div>
        <div className="thumbnails-container">
          <img src={steak} alt="single tag" />
          <p>steak</p>
        </div>
        <div className="thumbnails-container">
          <img src={streetfood} alt="single tag" />
          <p>street Food</p>
        </div>
        <div className="thumbnails-container">
          <img src={sushi} alt="single tag" />
          <p>sushi</p>
        </div>
        <div className="thumbnails-container">
          <img src={chinese} alt="single tag" />
          <p>chinese</p>
        </div>
      </div>
    </div>
  );
};

export default CategoriesList;
