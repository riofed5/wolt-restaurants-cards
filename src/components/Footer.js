import React, { Component } from "react";
import logo from "../images/logo.svg";
import appstore from "../images/appstore.svg";
import googleplay from "../images/googleplay.svg";

export default class Footer extends Component {
  render() {
    return (
      <div style={{ backgroundColor: "rgb(211, 210, 210)" }}>
        <div className="footer">
          <div className="footer-top">
            <span>
              <img src={logo} alt="Wolt" />
            </span>
          </div>
          <div className="footer-middle">
            <div className="footer-logo">
              <h3> Download the app</h3>
              <span>
                <img src={appstore} alt="apple store" />
              </span>
              <span>
                <img src={googleplay} alt="google play" />
              </span>
            </div>
            <div className="footer-detail">
              <h3>Let's do this together</h3>
              <p>For couriers</p>
              <p>For restaurant</p>
              <p>For companies</p>
              <p>Jobs</p>
            </div>
            <div className="footer-detail">
              <h3>Wolt links</h3>
              <p>Contact</p>
              <p>Media</p>
              <p>Support</p>
              <p>Blog</p>
            </div>
          </div>
          <div className="footer-bottom">
            <div className="footer-detail">
              <p>© Wolt 2014–2020</p>
            </div>
            <div className="footer-detail">
              <p>Terms and conditions</p>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
