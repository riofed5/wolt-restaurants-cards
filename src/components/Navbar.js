import React, { Component } from "react";
import { Link } from "react-router-dom";
import logo from "../images/logo.svg";

export default class Navbar extends Component {
  state = {
    languages: ["English", "Finnish", "Chinese", "German"],
    open: false
  };

  handleToggle = () => {
    this.setState({ open: !this.state.open });
  };

  render() {
    return (
      <div className="container-nav">
        <div className="nav-items">
          <div className="nav-left">
            <Link to="/discovery">
              <img src={logo} alt="Wolt" />
            </Link>
          </div>
          <div className="nav-middle">
            <input
              onChange={() => {
                console.log("it will change");
              }}
              value=""
              type="text"
              placeholder="Search"
              required
            />
          </div>
          <div className="nav-right" >
            <div className="nav-right-items">
              <input value="Sign Up" type="button" id="sign-up" />
              <input value="Log In" type="button" id="log-in" />
              <select id="languages">
                {this.state.languages.map((item, index) => {
                  return (
                    <option value={item} key={index}>
                      {item}
                    </option>
                  );
                })}
              </select>
            </div>
          </div>
          <div className="nav-more">
            <i
              className="fa fa-bars"
              aria-hidden="true"
              onClick={this.handleToggle}
            >
              <div
                style={{ display: `${this.state.open ? "block" : "none"}` }}
                className="options"
              >
                <div className="option-item"><span>Profile</span></div>
                <div className="option-item"><span>Languages</span></div>
                <div className="option-item-last"><span>Support</span></div>
              </div>
            </i>
          </div>
        </div>
      </div>
    );
  }
}
