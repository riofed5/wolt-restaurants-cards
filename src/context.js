import React, { Component } from "react";
import items from "./data";

const RecipeContext = React.createContext();

class RecipeProvider extends Component {
  state = {
    recipes: items,
    loading: true,
    sortedRecipes: items,
    layout: "grid",
    location: "Helsinki"
  };

  componentDidMount() {
    let tempRecipes = [...this.state.recipes];

    tempRecipes = tempRecipes.filter(
      recipe => recipe.city === this.state.location
    );

    this.setState({
      loading: false,
      sortedRecipes: tempRecipes
    });
  }

  handleLocation= ()=>{
    console.log('location');
  }

  handleSelectFilter = e => {
    const value = e.target.value;
    let tempRecipes = [...this.state.recipes];

    if (value === "A to Z") {
      tempRecipes = tempRecipes.sort((a, b) => a.name.localeCompare(b.name));
    } else if (value === "Z to A") {
      tempRecipes = tempRecipes.sort((a, b) => b.name.localeCompare(a.name));
    } else {
      tempRecipes= this.state.sortedRecipes;
    }
    this.setState({
      sortedRecipes: tempRecipes
    });
  };

  handleLayoutChange = value => {
    this.setState({
      layout: value
    });
  };

  render() {
    return (
      <RecipeContext.Provider
        value={{
          ...this.state,
          handleSelectFilter: this.handleSelectFilter,
          handleLayoutChange: this.handleLayoutChange
        }}
      >
        {this.props.children}
      </RecipeContext.Provider>
    );
  }
}

const RecipeConsumer = RecipeContext.Consumer;

export const withRecipeConsumer = Component => {
  const ConsumerWrapper = props => {
    return (
      <RecipeConsumer>
        {value => <Component {...props} context={value} />}
      </RecipeConsumer>
    );
  };
  return ConsumerWrapper;
};

export { RecipeConsumer, RecipeProvider, RecipeContext };
