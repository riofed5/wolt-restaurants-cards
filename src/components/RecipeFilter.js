import React from "react";
import { useContext } from "react";
import { RecipeContext } from "../context";

const RecipeFilter = () => {
  const context = useContext(RecipeContext);

  const selections = ["A TO Z", "Z TO A"];

  const { handleSelectFilter, handleLayoutChange, location } = context;

  return (
    <div className="container-filter">
      <div className="filter">
        <div className="filter-delivery-to">
          <span>Delivery to me</span>
          <span> in {location}</span>
        </div>
        <div className="filter-list">
          <select id="filter-select" onChange={handleSelectFilter}>
            <option selected="false" key={0}>
              SORT BY
            </option>
            {selections.map((item, index) => {
              return (
                <option value={item} key={index + 1}>
                  NAME MODEL: {item}
                </option>
              );
            })}
          </select>
          <div className="filter-button">
            <button
              className="filter-button-grid"
              onClick={() => handleLayoutChange("grid")}
            >
              <span>
                <i className="fa fa-th-large" aria-hidden="true" />
              </span>
            </button>
            <button
              className="filter-button-list"
              onClick={() => handleLayoutChange("line-by-line")}
            >
              <span>
                <i className="fa fa-list" aria-hidden="true" />
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default RecipeFilter;
