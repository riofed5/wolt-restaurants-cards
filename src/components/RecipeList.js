import React from "react";
import Recipe from "./Recipe";
import { useContext } from "react";
import { RecipeContext } from "../context";

const RecipeList = ({ recipes }) => {
  const deliveryRecipes = recipes.filter(recipe => recipe.online);

  const context = useContext(RecipeContext);

  const { layout } = context;
  console.log(layout);
  if (deliveryRecipes.length === 0) {
    return (
      <div className="empty-search">
        <h3>unfortunately no recipe matched your search</h3>
      </div>
    );
  }
  return (
    <div className=" recipeslist">
      <p className="title">Restaurants open &amp; online</p>
      <div
        className={
          layout === "grid" ? "recipeslist-center" : "recipeslist-center-layout"
        }
      >
        {deliveryRecipes.map((item, index) => {
          return <Recipe key={index} recipe={item} layout={layout} />;
        })}
      </div>
    </div>
  );
};

export default RecipeList;
