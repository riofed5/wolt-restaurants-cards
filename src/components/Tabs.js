import React from "react";
import { useContext } from "react";
import { RecipeContext } from "../context";
import { Link } from "react-router-dom";

const Tabs = () => {
  const context = useContext(RecipeContext);

  const { handleSelectTab, location } = context;

  const {address, city}= location
  
  return (
    <div className="container-tabs">
      <div className="tabs-items">
        <div className="tabs-left">
          <span>
            <i className="fa fa-map-marker" aria-hidden="true" />
          </span>
          <p>
            {address && address + "," } { city}
          </p>
        </div>
        <div className="tabs-right">
          <div className="tabs-right-items">
            <Link to="/discovery">
              <input
                type="button"
                value="Discovery"
                id="discovery"
                onClick={handleSelectTab}
              />
            </Link>
            <div className="vertical-divider"></div>
            <Link to="/delivery">
              <input
                type="button"
                value="Delivery"
                id="delivery"
                onClick={handleSelectTab}
              />
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};
export default Tabs;
