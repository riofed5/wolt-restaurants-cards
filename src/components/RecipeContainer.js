import React from 'react';
import Loading from '../components/Loading';
import RecipeFilter from '../components/RecipeFilter';
import RecipeList from '../components/RecipeList';
import { withRecipeConsumer} from '../context';


const RecipeContainer = ({context}) => {
    const {loading, sortedRecipes}= context;
    
    if(loading){
       return <Loading />
    }
    return (
        <>
            <RecipeFilter recipes={sortedRecipes}/>
            <RecipeList recipes={sortedRecipes} />
        </>
    )
}

export default withRecipeConsumer(RecipeContainer);
