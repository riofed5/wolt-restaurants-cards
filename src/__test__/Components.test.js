import CategoriesList from "../components/CategoriesList";
import Footer from "../components/Footer";
import Navbar from "../components/Navbar";
import Loading from "../components/Loading";

import Tabs from "../components/Tabs";
import RecipeContainer from "../components/RecipeContainer";
import RecipeFilter from "../components/RecipeFilter";
import RecipeList from "../components/RecipeList";

import renderer from "react-test-renderer";
import React, { useContext } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { RecipeContext, RecipeProvider } from "../context";

import Adapter from "enzyme-adapter-react-16";
import { mount, configure } from "enzyme";

/* Rendering test*/

it("renders categoriesList correctly", () => {
  const categoriesList = renderer.create(<CategoriesList />).toJSON();

  expect(categoriesList).toMatchSnapshot();
});

it("renders footer correctly", () => {
  const footer = renderer.create(<Footer />).toJSON();

  expect(footer).toMatchSnapshot();
});

it("renders navbar correctly", () => {
  const navbar = renderer
    .create(
      <Router>
        <Navbar />
      </Router>
    )
    .toJSON();

  expect(navbar).toMatchSnapshot();
});

it("renders Loading correctly", () => {
  const loading = renderer.create(<Loading />).toJSON();

  expect(loading).toMatchSnapshot();
});
/* End rendering test*/

/*----------------------------------------------------------------------------*/

/* Functionality test */

//Test for handle Filter
configure({ adapter: new Adapter() });
describe("Sort items ", () => {
  
  //sort name from A to Z
  it("by name ascending", () => {
    const TestComponent = () => {
      const context = useContext(RecipeContext);

      let { handleSelectFilter, sortedRecipes } = context;

      return (
        <>
          <select onChange={handleSelectFilter}>
            <option value="A to Z" key={1}>
              Name Ascending
            </option>
          </select>
          <div data-testid="value">
            {sortedRecipes.map(item => item.name).toString()}
          </div>
        </>
      );
    };

    const wrapper = mount(
      <RecipeProvider>
        <TestComponent />
      </RecipeProvider>
    );

    wrapper.find("select").simulate("change");
    expect(wrapper.find('[data-testid="value"]').text()).toEqual(
      "Arnolds Forum,Base Camp Yliopistonkatu,Biáng! Citycenter,Burger King Mannerheimintie,Classic Pizza Restaurant Forum,Classic Pizza Restaurant Stockmann,Döner Harju City,Eat Poke HKI,Elmo Sport Bistro,Fafa's Citykäytävä,Fafa's Mikonkatu,Fafa's Sokos,Fazer Café Forum,FLQ Kaisaniemi,FreshStop Eliel,Friends & Brgrs Helsinki,Fuku Helsinki,Grande Grill & Green,Haiku Sushi,Hanko Sushi Forum,Hanko Sushi Stockmann,Happy Waffle Helsinki,Hey Poke Citycenter,Huacheng,Jufu Rautatientori,Jungle Juice Bar Forum,Kippo,La Torrefazione Lasipalatsi,Luckiefun's Kaivokatu,Mashiro Postitalo,McDonald's Forum Katutaso,Momotoko Citycenter,MorriSon's Helsinki,O'Learys Bakers,Picnic Forum Kukontori,Picnic Yliopistonkatu,Pizza Hut Citykäytävä,Pizza Hut Forum,Pizzeria & Spagetteria MorriSon's,Sandro Citykäytävä,Santa Fé,Siipiweikot Helsinki,Social Burgerjoint Citycenter,Soma Food, Kaivokatu,Suburritos,Sushibar + Wine City,Taco Bell Baker's,Vapiano Mikonkatu,Woolshed – Australian Gastropub,Zhonghua"
    );
  });

  //sort name from Z to A

  it("by name descending", () => {
    const TestComponent = () => {
      const context = useContext(RecipeContext);

      let { handleSelectFilter, sortedRecipes } = context;

      return (
        <>
          <select onChange={handleSelectFilter}>
            <option value="Z to A" key={1}>
              Name Descending
            </option>
          </select>
          <div data-testid="value">
            {sortedRecipes.map(item => item.name).toString()}
          </div>
        </>
      );
    };

    const wrapper = mount(
      <RecipeProvider>
        <TestComponent />
      </RecipeProvider>
    );

    wrapper.find("select").simulate("change");
    expect(wrapper.find('[data-testid="value"]').text()).toEqual(
      "Zhonghua,Woolshed – Australian Gastropub,Vapiano Mikonkatu,Taco Bell Baker's,Sushibar + Wine City,Suburritos,Soma Food, Kaivokatu,Social Burgerjoint Citycenter,Siipiweikot Helsinki,Santa Fé,Sandro Citykäytävä,Pizzeria & Spagetteria MorriSon's,Pizza Hut Forum,Pizza Hut Citykäytävä,Picnic Yliopistonkatu,Picnic Forum Kukontori,O'Learys Bakers,MorriSon's Helsinki,Momotoko Citycenter,McDonald's Forum Katutaso,Mashiro Postitalo,Luckiefun's Kaivokatu,La Torrefazione Lasipalatsi,Kippo,Jungle Juice Bar Forum,Jufu Rautatientori,Huacheng,Hey Poke Citycenter,Happy Waffle Helsinki,Hanko Sushi Stockmann,Hanko Sushi Forum,Haiku Sushi,Grande Grill & Green,Fuku Helsinki,Friends & Brgrs Helsinki,FreshStop Eliel,FLQ Kaisaniemi,Fazer Café Forum,Fafa's Sokos,Fafa's Mikonkatu,Fafa's Citykäytävä,Elmo Sport Bistro,Eat Poke HKI,Döner Harju City,Classic Pizza Restaurant Stockmann,Classic Pizza Restaurant Forum,Burger King Mannerheimintie,Biáng! Citycenter,Base Camp Yliopistonkatu,Arnolds Forum"
    );
  });
});

/* End functionality test */
