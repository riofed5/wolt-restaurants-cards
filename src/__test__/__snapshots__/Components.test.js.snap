// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loading correctly 1`] = `
<div
  className="loading"
>
  <h4>
    rooms is loading
  </h4>
  <img
    alt=""
    src="loading-gear.gif"
  />
</div>
`;

exports[`renders categoriesList correctly 1`] = `
<div
  className="categories"
>
  <div
    className="categories-title"
  >
    <span>
      Categories
    </span>
  </div>
  <div
    className="categories-center"
  >
    <div
      className="thumbnails-container"
    >
      <img
        alt="single tag"
        src="cat_kebab.jpg"
      />
      <p>
        kebab
      </p>
    </div>
    <div
      className="thumbnails-container"
    >
      <img
        alt="single tag"
        src="cat_ramen.jpg"
      />
      <p>
        ramen
      </p>
    </div>
    <div
      className="thumbnails-container"
    >
      <img
        alt="single tag"
        src="cat_steak.jpg"
      />
      <p>
        steak
      </p>
    </div>
    <div
      className="thumbnails-container"
    >
      <img
        alt="single tag"
        src="cat_street_food.jpg"
      />
      <p>
        street Food
      </p>
    </div>
    <div
      className="thumbnails-container"
    >
      <img
        alt="single tag"
        src="cat_sushi.jpg"
      />
      <p>
        sushi
      </p>
    </div>
    <div
      className="thumbnails-container"
    >
      <img
        alt="single tag"
        src="cat_chinese.jpg"
      />
      <p>
        chinese
      </p>
    </div>
  </div>
</div>
`;

exports[`renders footer correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "rgb(211, 210, 210)",
    }
  }
>
  <div
    className="footer"
  >
    <div
      className="footer-top"
    >
      <span>
        <img
          alt="Wolt"
          src="logo.svg"
        />
      </span>
    </div>
    <div
      className="footer-middle"
    >
      <div
        className="footer-logo"
      >
        <h3>
           Download the app
        </h3>
        <span>
          <img
            alt="apple store"
            src="appstore.svg"
          />
        </span>
        <span>
          <img
            alt="google play"
            src="googleplay.svg"
          />
        </span>
      </div>
      <div
        className="footer-detail"
      >
        <h3>
          Let's do this together
        </h3>
        <p>
          For couriers
        </p>
        <p>
          For restaurant
        </p>
        <p>
          For companies
        </p>
        <p>
          Jobs
        </p>
      </div>
      <div
        className="footer-detail"
      >
        <h3>
          Wolt links
        </h3>
        <p>
          Contact
        </p>
        <p>
          Media
        </p>
        <p>
          Support
        </p>
        <p>
          Blog
        </p>
      </div>
    </div>
    <div
      className="footer-bottom"
    >
      <div
        className="footer-detail"
      >
        <p>
          © Wolt 2014–2020
        </p>
      </div>
      <div
        className="footer-detail"
      >
        <p>
          Terms and conditions
        </p>
      </div>
      <div
        className="footer-detail"
      >
        <p>
          Terms and conditions
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders navbar correctly 1`] = `
<div
  className="container-nav"
>
  <div
    className="nav-items"
  >
    <div
      className="nav-left"
    >
      <a
        href="/discovery"
        onClick={[Function]}
      >
        <img
          alt="Wolt"
          src="logo.svg"
        />
      </a>
    </div>
    <div
      className="nav-middle"
    >
      <input
        onChange={[Function]}
        placeholder="Search"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="nav-right"
    >
      <div
        className="nav-right-items"
      >
        <input
          id="sign-up"
          type="button"
          value="Sign Up"
        />
        <input
          id="log-in"
          type="button"
          value="Log In"
        />
        <select
          id="languages"
        >
          <option
            value="English"
          >
            English
          </option>
          <option
            value="Finnish"
          >
            Finnish
          </option>
          <option
            value="Chinese"
          >
            Chinese
          </option>
          <option
            value="German"
          >
            German
          </option>
        </select>
      </div>
    </div>
    <div
      className="nav-more"
    >
      <i
        aria-hidden="true"
        className="fa fa-bars"
        onClick={[Function]}
      >
        <div
          className="options"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="option-item"
          >
            <span>
              Profile
            </span>
          </div>
          <div
            className="option-item"
          >
            <span>
              Languages
            </span>
          </div>
          <div
            className="option-item-last"
          >
            <span>
              Support
            </span>
          </div>
        </div>
      </i>
    </div>
  </div>
</div>
`;
